wrappers/gin/gin.go

```diff
@@ -0,0 +1,90 @@
+package epsagongin
+
+import (
+	// "github.com/epsagon/epsagon-go/internal"
+	// "github.com/epsagon/epsagon-go/protocol"
+	"fmt"
+	"net/http"
+
+	"github.com/epsagon/epsagon-go/epsagon"
+	"github.com/epsagon/epsagon-go/tracer"
+	"github.com/gin-gonic/gin"
+)
+
+// TracerKey is the key of the epsagon tracer in the gin.Context Keys map passed to the handlers
+const TracerKey = "EpsagonTracer"
+
+// GinRouterWrapper is an epsagon instumentation wrapper for gin.RouterGroup
+type GinRouterWrapper struct {
+	gin.IRouter
+	Hostname string
+	Config   *epsagon.Config
+}
+
+func wrapGinHandler(handler gin.HandlerFunc, hostname string, relativePath string, config *epsagon.Config) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		if config == nil {
+			config = &epsagon.Config{}
+		}
+		wrapperTracer := tracer.CreateTracer(&config.Config)
+		wrapperTracer.Start()
+		defer wrapperTracer.Stop()
+
+		if c.Keys == nil {
+			c.Keys = make(map[string]interface{})
+		}
+		c.Keys[TracerKey] = wrapperTracer
+		wrapper := epsagon.WrapGenericFunction(
+			handler,
+			config,
+			wrapperTracer,
+			false,
```

Comment:
If the wrapped handler will wrap internal requests events - they obviously won't be collected. I guess we should document it somewhere (probably in the README)